### PR TITLE
fix/added-guide-to-explain-how-biome-behaves-with-new-files

### LINKED ISSUE #587: 📎 Add a guide to explain how biome behaves with *known* files
### Description

This is a follow-up of the discussion that happened in https://github.com/biomejs/biome/issues/237

The task is to create a new guide page called "Biome's behaviour" (or something similar). The purpose of this page is to list all those particular behaviours/assumptions that Biome does that aren't currently documented, or they are but in the wrong section.

Here's the list of things we should do:
1. Move the following paragraphs in this new page:

https://github.com/biomejs/biome/blob/e6c12c0fed2a1b1662f2f228d369917dd58b7a5c/website/src/content/docs/reference/configuration.mdx#L13-L65

2. Explain that `package.json`, `package-lock.json`, `tsconfig.json` and `jsconfig.json` are currently ignored by Biome because the tools that own these files usually apply their formatting/style, and Biome doesn't want to disrupt them. This means that **no diagnostics** will be ever emitted by Biome for those files.
3. Add a heading that explains the "known" files. These files are listed here:

https://github.com/biomejs/biome/blob/e6c12c0fed2a1b1662f2f228d369917dd58b7a5c/crates/biome_service/src/file_handlers/mod.rs#L48-L64

	And we should explain that when Biome analyses these files, they are parsed as **JSON files** with `json.parser.allowComments` and `json.parser.allowTrailingCommas` set to `true`. This is because editor tools like VSCode treat them like this.

	As a bonus, we should remove `tsconfig.json` and `jsconfig.json` from that list (in the rust file)

### PR DESCRIPTION
**Pull request by TheShiveshNetwork**
## Summary

Fix #587 
This PR adds the file `BIOME_BEHAVIOURS.md` for the guide to explain how biome behaves with new files.

### CODE REVIEW

**Review by ematipico — CHANGES_REQUESTED** [review]
Here's some early feedback:

- we want to copy all this content
	https://github.com/biomejs/biome/blob/4464e561296d1e753b4873697d1818fe1a5f2605/website/src/content/docs/reference/configuration.mdx#L11-L65
- we need to remove the content from `configuration.mdx`
	
- the page should be called `biome-behaviour.mdx`
- the page doesn't have the correct frontmatter, like this one:
	https://github.com/biomejs/biome/blob/4464e561296d1e753b4873697d1818fe1a5f2605/website/src/content/docs/guides/getting-started.mdx#L1-L4
- the page needs to be added to the sidebar
	https://github.com/biomejs/biome/blob/4464e561296d1e753b4873697d1818fe1a5f2605/website/astro.config.ts#L171-L181

**Comment by ematipico** [comment]
@TheShiveshNetwork please make sure to [website locally](https://github.com/biomejs/biome/blob/main/CONTRIBUTING.md#website-development), because with the current changes, the page **doesn't** appear in the website.

**Comment by TheShiveshNetwork** [comment]
@ematipico alright! Working on it.

**Comment by TheShiveshNetwork** [comment]
@ematipico Just pushed a commit, tell me if I missed something. I am ready for further improvisations.

**Comment by TheShiveshNetwork** [comment]
And yeah, can you help me with the project setup and how do I run this project locally, because when I use the command
`just install-tools`
I get
`error: Recipe 'install-tools' could not be run because just could not find the shell: program not found`.

**Comment by ematipico** [comment]
That's weird, I have never seen that error. Did you install rust on your machine?

**Comment by TheShiveshNetwork** [comment]
@ematipico Sorry to disturb you again. But can you check this now?
I removed the dependancy
`"@biomejs/wasm-web": "link:../packages/@biomejs/wasm-web",`
And commented the places where it was used.

**Comment by ematipico** [comment]
Hey @TheShiveshNetwork , I think there was a misunderstanding.

You should **not** remove the things I asked you to comment, I suggested you to that **locally** on your machine so you could start the dev server of the website on your machine.

File: biome-behaviour.mdx [hunk 1]
```diff
@@ -0,0 +1,83 @@
+---
+title: Getting Started
+category: reference
+description: Learn how to set up a new project with Biome.
+---
```
Comment (ematipico) [inline biome-behaviour.mdx:5]: This isn't correct, it should not be copied. I just showed you **an example**, but here you have:

```
---
title: How Biome works
description: Understand how the Biome toolchain works
---
```

**Comment by TheShiveshNetwork** [comment]
Sorry from my end, I misunderstood it! I'll make another commit.

**Comment by ematipico** [comment]
Thank you @TheShiveshNetwork , I will take it from here

**Comment by TheShiveshNetwork** [comment]
Thanks @ematipico it was a great experience working on this, with your suggestions and help.

**Pull request merged**